test/assert.js

```diff
@@ -182,6 +183,29 @@ test('.throws()', function (t) {
 	t.end();
 });
 
+test('.throws() becomes the thrown Error', function (t) {
+	var expected = new Error();
+	var actual = assert.throws(function () {
+		throw expected;
+	});
+
+	assert.same(actual, expected);
+
+	t.end();
+});
+
+test('.throws() becomes the rejection reason of promise', function (t) {
+	var expected = new Error();
+	var exec = new Promise(function (resolve) {
+		resolve(assert.throws(Promise.reject(expected)));
+	});
+
+	exec.then(function (actual) {
```

Comment:
This should work:

```js
assert.throws(Promise.reject(expected)).then(function (actual) {
```

`assert.throws()` already returns a promise because you pass one rather than a function. That promise fulfils with the rejection reason.